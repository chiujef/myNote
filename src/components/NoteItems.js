import PropTypes from 'prop-types';
import React, {Fragment} from 'react';
import {NoteItem} from '@components';

const NoteItems = ({
  noteItems,
  noteItemSelected,
  onNoteItemCompletedToggle,
  onNoteItemChangeText,
  onNoteItemPress,
  onNoteItemDeletePress,
}) => {
  return (
    <Fragment>
      {noteItems.map(noteItem => {
        let isSelected = noteItemSelected === noteItem.id;
        return (
          <NoteItem
            key={noteItem.id}
            noteItem={noteItem}
            isSelected={isSelected}
            onCompletedToggle={onNoteItemCompletedToggle}
            onChangeText={onNoteItemChangeText}
            onTextPress={onNoteItemPress}
            onDeletePress={onNoteItemDeletePress}
          />
        );
      })}
    </Fragment>
  );
};

NoteItems.propTypes = {
  noteItems: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      text: PropTypes.string,
      completed: PropTypes.bool,
    }),
  ),
  noteItemSelected: PropTypes.number,
  onNoteItemCompletedToggle: PropTypes.func,
  onNoteItemChangeText: PropTypes.func,
  onNoteItemPress: PropTypes.func,
  onNoteItemDeletePress: PropTypes.func,
};

export default NoteItems;
