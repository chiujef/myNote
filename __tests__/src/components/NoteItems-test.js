import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render} from 'react-native-testing-library';

import NoteItems from '@components/NoteItems';

describe('NoteItems', () => {
  let noteItems;

  beforeEach(() => {
    noteItems = [
      {
        id: 1,
        text: 'tent',
        completed: false,
      },
      {
        id: 2,
        text: 'tent',
        completed: true,
      },
    ];
  });

  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = renderer
        .create(<NoteItems noteItems={noteItems} noteItemSelected={-1} />)
        .toJSON();
      expect(component).toMatchSnapshot();
    });

    it('should render all the texts', () => {
      const {getAllByText} = render(
        <NoteItems noteItems={noteItems} noteItemSelected={-1} />,
      );

      expect(getAllByText('tent').length).toBe(2);
    });
  });
});
