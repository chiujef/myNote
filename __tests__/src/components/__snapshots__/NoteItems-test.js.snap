// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItems Rendering should match to snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="checkbox"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#bfbfbf",
                "fontSize": 24,
              },
              Object {
                "minWidth": 24,
              },
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        clickable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        tent
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 0.25,
          "justifyContent": "center",
          "margin": 1,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="checkbox"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#2089dc",
                "fontSize": 24,
              },
              Object {
                "minWidth": 24,
              },
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        clickable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "textDecorationLine": "line-through",
            "textDecorationStyle": "solid",
          }
        }
      >
        tent
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 0.25,
          "justifyContent": "center",
          "margin": 1,
        }
      }
    />
  </View>,
]
`;
